resources/index.js

```diff
@@ -80,18 +80,18 @@ const map = {
  * @private
  */
 function registerAll(Shopify) {
-  Object.keys(map).forEach((prop) => {
-    Object.defineProperty(Shopify.prototype, prop, {
+  Object.entries(map).forEach(([resourceName, importResource]) => {
+    Object.defineProperty(Shopify.prototype, resourceName, {
       configurable: true,
       get: function get() {
-        const resource = require(`./${map[prop]}`);
+        const resource = importResource();
```

Comment:
I don't understand the difference. What bundler issue is this fixing? And why a wrapper around `require()` fixes it? Also, can something like this be used

```js
function importResource(name) {
  return require(name);
}
```

instead of defining a dedicated function for each resource?

Reply:
Bundlers work by traversing code, resolving imports/requires until there are none left, adding them to a bundle as it goes

The problem we are facing when bundling is due to the use of a dynamic token when importing`${map[prop]}`. Bundlers aren't smart enough to know the context and how to resolve that import, especially when in a loop

The fix is to explicitly define the lazy imports (or require) in a way that can be analysed statically by bundlers. Which means we need to define imports like so `require('./payout')` so that bundlers can successfully resolve them

While it's more verbose the outcome is exactly the same, lazy imports at runtime. But with the benefit of being friendly for bundling :)

Reply:
> Bundlers aren't smart enough to know the context and how to resolve that import, especially when in a loop

What bundler specifically? I think it works with some of them.

Reply:
You're right, it does work with webpack IIRC.

I am using esbuild, which doesn't support this pattern of dynamic imports

Do you have any concerns with this change? It is 100% backwards compatible with identical behaviour, but solves a problem a few people have encountered